addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/ThirdGeneration/ThirdGenerationHandler.java

```diff
@@ -0,0 +1,767 @@
+
+        /*
+         * Start the initialization
+         */
+        authenticate();
+
+        /*
+         * Start the update scheduler as configured
+         */
+        refreshScheduler = scheduler.scheduleWithFixedDelay(() -> {
+            updateChannelValues();
+        }, 10, config.refreshInternalInSeconds, TimeUnit.SECONDS);
+    }
+
+    /**
+     * The API supports the resolution of multiple values at a time
+     *
+     * Therefore this methods builds one request to gather all information for the current inverter.
+     * The list contains all channels as defined in {@link ThirdGenerationMappingInverterToChannel} for the
+     * current inverter
+     *
+     */
+    void updateChannelValues() {
+        Map<String, List<ThirdGenerationChannelMappingToWebApi>> channelList = ThirdGenerationMappingInverterToChannel
+                .getModuleToChannelsMappingForInverter(inverterType);
+        JsonArray updateMessageJsonArray = new JsonArray();
+        /*
+         * Build the message to send to the inverter
+         */
+        for (String moduleId : channelList.keySet()) {
+            JsonObject moduleJsonObject = new JsonObject();
+            moduleJsonObject.addProperty("moduleid", moduleId);
+
+            JsonArray processdataNames = new JsonArray();
+            for (ThirdGenerationChannelMappingToWebApi processdata : channelList.get(moduleId)) {
+                processdataNames.add(processdata.processdataId);
+            }
+            moduleJsonObject.add("processdataids", processdataNames);
+            updateMessageJsonArray.add(moduleJsonObject);
+        }
+        /*
+         * Send the API request to get values for all channels
+         * Note: If one does not exist, all fail
+         */
+        ContentResponse updateMessageContentResponse;
+        try {
+            updateMessageContentResponse = executeHttpPost(PROCESSDATA, updateMessageJsonArray, sessionId);
+            if (updateMessageContentResponse.getStatus() == 404) {
+                /*
+                 * Module not found
+                 */
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        COMMUNICATION_ERROR_INCOMPATIBLE_DEVICE);
+                return;
+            }
+            if (updateMessageContentResponse.getStatus() == 503) {
+                /*
+                 * Communication error (e.g. during initial boot of the SCB)
+                 */
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                        COMMUNICATION_ERROR_HTTP);
+                return;
+            }
+            if (updateMessageContentResponse.getStatus() == 401) {
+                /*
+                 * session not valid (timed out?)
+                 */
+                logger.info("Session expired");
+                try {
+                    authenticate();
+                } catch (Exception e) {
```

Comment:
Don't catch exception here. I don't know why you added it here. But if any catch `RuntimeException` and certainly do log the exception (to debug), if there is a bug you'll never find out otherwise.

Reply:
The first shot (using the Apache http Client) was way better in exception handling.
"Unfortunatly" @wborn asked me to switch to jetty
The start-method on the jetty implementation throws the exception base class and not any specific … any idea how to work around this? (see https://www.eclipse.org/jetty/javadoc/current/org/eclipse/jetty/util/component/ContainerLifeCycle.html#start-org.eclipse.jetty.util.component.LifeCycle- for details; the jetty http Client derives from this class)

Reply:
If an external method throws it you need to handle it. Then you most likely will need to handle `InterruptedException` separately and make sure interrupt is set again.

In this specific case you wrap it around your `authenticate` method which doesn't throw any checked exceptions at all so it can be removed here.

We try to limit the amount of external dependencies because each dependency adds more bytes to the total download size and memory usage and to keep things consistent. As the jetty is part of the core it is the preferred way. I would guess you only have the exception problems because you create the http client yourself (and start and stop it). With using the provided one you don't need to start and stop, So I would assume you won't have to handle the generic exception.

Reply:
I will review this issue after switching the httpClientFactory. Hopefully this will fix my Exception-base-class problem with it and I can switch back to doing smooth exception handling :-)

Reply:
The Change to httpClientFactory had the desired effect.
Exception handling improved :-)